website/blog/how-to-migrate-an-ios-app-to-bazel.md

```diff
@@ -0,0 +1,1470 @@
+# commands to resolve external dependencies. This results in a `.build` file
+# being created.
+# NOTE: Swift package manager is not used to build any of the external packages.
+# The `.build` directory should be ignored. Be sure to configure your source
+# control to ignore it (i.e., add it to your `.gitignore`).
+# gazelle:exclude .build
+
+# This declaration builds a Gazelle binary that incorporates all of the Gazelle
+# plugins for the languages that you use in your workspace. In this example, we
+# are only listing the Gazelle plugin for Swift from
+# rules_swift_package_manager.
+gazelle_binary(
+    name = "gazelle_bin",
+    languages = [
+        "@rules_swift_package_manager//gazelle",
+    ],
+)
+
+# This macro defines two targets: `swift_update_pkgs` and
+# `swift_update_pkgs_to_latest`.
+#
+# The `swift_update_pkgs` target should be run whenever the list of external
+# dependencies is updated in the `Package.swift`. Running this target will
+# populate the `swift_deps.bzl` with `swift_package` declarations for all of the
+# direct and transitive Swift packages that your project uses.
+#
+# The `swift_update_pkgs_to_latest` target should be run when you want to
+# update your Swift dependencies to their latest eligible version.
+swift_update_packages(
+    name = "swift_update_pkgs",
+    gazelle = ":gazelle_bin",
+    generate_swift_deps_for_workspace = False,
+    patches_yaml = "patches/swiftpkgs.yaml",
+    update_bzlmod_stanzas = True,
+)
+
+# This target updates the Bazel build files for your project. Run this target
+# whenever you add or remove source files from your project.
+gazelle(
+    name = "update_build_files",
+    gazelle = ":gazelle_bin",
+)
+```
+
+:::note
+
+Because of how **rules_apple** handles app icon assets
+(though possibly because of a bug),
+we need to [patch][rspm-patch] the **Tabman** package to not include test assets.
+Our patches are defined in the `patches/swiftpkgs.yaml` file.
+
+[rspm-patch]: https://github.com/cgrindel/rules_swift_package_manager/blob/v0.28.0/docs/patch_swift_package.md
+
+:::
+
+To generate the `swift_deps_index.json` file,
+and/or update the `MODULE.bazel` `use_repo` stanza,
+run the `//:swift_update_pkgs` target:
+
+```shellsession
+$ bazel run //:swift_update_pkgs
+INFO: Invocation ID: f2427bde-865b-43b3-bd3c-6a8489387f00
+INFO: Analyzed target //:swift_update_pkgs (148 packages loaded, 11707 targets configured).
+INFO: Found 1 target...
+Target //:swift_update_pkgs up-to-date:
+  bazel-bin/swift_update_pkgs-runner.bash
+  bazel-bin/swift_update_pkgs
+INFO: Elapsed time: 1.724s, Critical Path: 0.07s
+INFO: 1 process: 1 internal.
+INFO: Build completed successfully, 1 total action
+INFO: Running command line: bazel-bin/swift_update_pkgs
+```
+
+At [this commit][commit-rspm] Swift packages have been integrated for the Mastodon iOS project.
+
+[commit-rspm]: https://github.com/brentleyjones/mastodon-ios/commit/06c9d960622245ecf5add45692e3acf410ae5729
```

Comment:
Perhaps a link to a Github issue for `# Work around rules_swift_package_manager sandboxing issue`, if it's not too much trouble.